src/Init/System/IO.lean

```diff
@@ -594,6 +583,26 @@ end System.FilePath
 
 namespace IO
 
+namespace FS
+
+def readBinFile (fname : FilePath) : IO ByteArray := do
+  -- Requires metadata so defined after metadata
+  let mdata ← fname.metadata
+  let size := mdata.byteSize.toUSize
+  if size > 0 then
+    let handle ← IO.FS.Handle.mk fname .read
+    handle.read mdata.byteSize.toUSize
```

Comment:
Is this based on any prior implementation? What if the file is `/dev/stdin`?

Reply:
This is based on Rust's [`read_to_string`](https://doc.rust-lang.org/stable/src/std/fs.rs.html#302-312):
```rs
pub fn read_to_string<P: AsRef<Path>>(path: P) -> io::Result<String> {
    fn inner(path: &Path) -> io::Result<String> {
        let mut file = File::open(path)?;
        let size = file.metadata().map(|m| m.len() as usize).ok();
        let mut string = String::new();
        string.try_reserve_exact(size.unwrap_or(0))?;
        io::default_read_to_string(&mut file, &mut string, size)?;
        Ok(string)
    }
    inner(path.as_ref())
}
```

Reply:
As for `/dev/stdin` it does have a size:
```
λ ls -la /dev/stdin
lrwxrwxrwx. 1 root root 15 Jul 29 21:02 /dev/stdin -> /proc/self/fd/0
```
So this implementation would at least not do something completely crazy. That said I think the advice is if you are reading from something like /dev/stdin that can evolve over time the advice is to basically not use `readBinFile`/`readFile`.

Reply:
I don't think that's quite the same, [`default_read_to_string`](https://stdrs.dev/nightly/x86_64-unknown-linux-gnu/src/std/io/mod.rs.html#432-447) looks like it might only use the size as a hint (and passes it as an option)